internal/builder/dynamic_mappings.go

```diff
@@ -0,0 +1,256 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package builder
+
+import (
+	_ "embed"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/elastic/elastic-package/internal/formatter"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/packages/buildmanifest"
+)
+
+//go:embed _static/ecs_mappings.json
+var staticEcsMappings string
+
+type ecsTemplates struct {
+	Mappings struct {
+		Properties       map[string]interface{}   `yaml:"properties"`
+		DynamicTemplates []map[string]interface{} `yaml:"dynamic_templates"`
+	} `yaml:"mappings"`
+}
+
+func addDynamicMappings(packageRoot, destinationDir string) error {
+	bm, ok, err := buildmanifest.ReadBuildManifest(packageRoot)
+	if err != nil {
+		return errors.Wrap(err, "can't read build manifest")
+	}
+	if !ok {
+		logger.Debugf("Build manifest hasn't been defined for the package")
+		return nil
+	}
+	if !bm.ImportCommonDynamicMappings() {
+		logger.Debugf("Package doesn't have to import common dynamic mappings")
```

Comment:
I would log on the opposite, when the package is importing mappings :slightly_smiling_face: